### PR TITLE
DPL Analysis: add option to artificially slow down AOD reader

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

**Comment by ktf** [comment]
@costing @jgrosseo this should take care of it. Now testing on hyperloop.

**Comment by ktf** [comment]
Tested on hyperloop in the current incarnation. Works as advertised.

File: Framework/AnalysisSupport/src/AODJAlienReaderHelpers.cxx [hunk 1]
```diff
@@ -278,6 +281,19 @@ AlgorithmSpec AODJAlienReaderHelpers::rootFileReaderCallback()
...
         first = false;
       }
+      int64_t stopSize = totalSizeCompressed;
+      int64_t bytesDelta = stopSize - startSize;
+      int64_t stopTime = uv_hrtime();
+      float currentDelta = float(stopTime - startTime) / 1000000000; // in s
+      if (ceil(maxRate) > 0.) {
+        float extraTime = (bytesDelta / 1000000 - currentDelta * maxRate) / maxRate;
+        // We only sleep if we read faster than the max-read-rate.
+        LOGP(info, "Read {} MB in {} s. Sleeping for {} seconds to stay within {} MB/s limit.", bytesDelta / 1000000, currentDelta, extraTime, maxRate);
```
Comment (jgrosseo) [inline Framework/AnalysisSupport/src/AODJAlienReaderHelpers.cxx]: Shouldn't we put the log only when extraTime is actually > 0 ?

**Comment by jgrosseo** [comment]
Thanks! I can check this out when I am back next week.

File: Framework/AnalysisSupport/src/AODJAlienReaderHelpers.cxx (hunk 1 above)
Comment (ktf) [reply Framework/AnalysisSupport/src/AODJAlienReaderHelpers.cxx]: fair enough.

**Pull request merged**